examples/C++/LifespanQoSExample/LifespanPublisher.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file LifespanPublisher.cpp
+ *
+ */
+
+#include "LifespanPublisher.h"
+#include <fastrtps/participant/Participant.h>
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/PublisherAttributes.h>
+#include <fastrtps/publisher/Publisher.h>
+#include <fastrtps/Domain.h>
+#include <fastrtps/utils/eClock.h>
+
+#include <thread>
+
+using namespace eprosima::fastrtps;
+using namespace eprosima::fastrtps::rtps;
+
+LifespanPublisher::LifespanPublisher()
+    : mp_participant(nullptr)
+    , mp_publisher(nullptr)
+{
+}
+
+bool LifespanPublisher::init(uint32_t lifespan_ms)
+{
+    m_Hello.index(0);
+    m_Hello.message("Lifespan");
+
+    ParticipantAttributes PParam;
+    PParam.rtps.builtin.domainId = 0;
+    PParam.rtps.builtin.leaseDuration = c_TimeInfinite;
+    PParam.rtps.setName("Participant_pub");
+    mp_participant = Domain::createParticipant(PParam);
+    if( mp_participant == nullptr )
+    {
+        return false;
+    }
+
+    Domain::registerType(mp_participant,&m_type);
+
+    PublisherAttributes Wparam;
+    Wparam.topic.topicKind = NO_KEY;
+    Wparam.topic.topicDataType = "Lifespan";
+    Wparam.topic.topicName = "LifespanTopic";
+    Wparam.topic.historyQos.kind = KEEP_ALL_HISTORY_QOS;
+    Wparam.topic.historyQos.depth = 30;
+    Wparam.qos.m_reliability.kind = RELIABLE_RELIABILITY_QOS;
```

Comment:
Even it is the default value, I will add the QoS durability to `TRANSIENT_LOCAL`. So the user could figure out that is the reason the example can remove samples from history after being sent.

Reply:
Added